### PR TITLE
Parser: fix issues with `end` marker syntax for val

### PR DESCRIPTION
**Pull request by dos65**

### CODE REVIEW

**Review by kpbochenek — APPROVED** [review]
run scalafmt before merging, quick fix thanks :+1:

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -1192,7 +1192,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       token.is[Interpolation.End] || token.is[Xml.End] ||
       token.is[KwReturn] || token.is[KwThis] || token.is[KwType] ||
       token.is[RightParen] || token.is[RightBracket] || token.is[RightBrace] ||
-      token.is[Underscore] || token.is[Ellipsis] || token.is[Unquote]
+      token.is[Underscore] || token.is[Ellipsis] || token.is[Unquote] ||
+      token.prev.is[EndMarkerIntro]
```
Comment (kitbellew) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1196]: does this change apply to all end markers? should we undo @tgodzik's special-case handling for match? #2276

Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1196]: Just checked and we still need that one.

Comment (dos65) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1196]: These tho cases with `match` and `val` are different

**Pull request merged**